core/src/main/java/org/apache/iceberg/BaseIncrementalScan.java

```diff
@@ -35,6 +35,19 @@ protected BaseIncrementalScan(
   protected abstract CloseableIterable<T> doPlanFiles(
       Long fromSnapshotIdExclusive, long toSnapshotIdInclusive);
 
+  @Override
+  public ThisT fromSnapshotInclusive(long fromSnapshotId, String branch) {
+    Preconditions.checkArgument(
+        table().snapshot(fromSnapshotId) != null,
+        "Cannot find the starting snapshot: %s",
+        fromSnapshotId);
+    Preconditions.checkArgument(
+        table().snapshot(branch) != null, "Cannot find the branch: %s", branch);
+    TableScanContext newContext =
```

Comment:
The validation that the fromSnapshotId is an ancestor needs to happen before the context is constructed, same for the exclusive case.

Reply:
Done.